### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by nubela**
Allow labels for training models to use more than just capped letters, but numbers and small letters as well.

Also fixed a bug with socket shutdown in finally, rather than in try. (shutdown() can fail too)

### CODE REVIEW

File: ner/client.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
 
 #regex patterns for various tagging options for entity parsing
-SLASHTAGS_EPATTERN  = re.compile(r'(.+?)/([A-Z]+)?\s*')
+SLASHTAGS_EPATTERN  = re.compile(r'(.+?)/([0-Z]+)?\s*')
```
Comment (dat) [inline ner/client.py]: Please use [0-9a-zA-Z] for clarity sake.

Comment (nubela) [inline ner/client.py]: Good point. Let me fix it.

On Fri, Feb 8, 2013 at 12:35 AM, dat hoang notifications@github.com wrote:

> [0-9a-zA-Z]

Steven Goh,
_M:_ +65 96634246
Founder, Spawt.

_Spawt is the easiest way to discover great things happening at places
nearby. *
*Download Spawt for your iPhone now at http://dl.spawt.me_

File: ner/utils.py [hunk 2]
```diff
@@ -13,9 +13,10 @@ def tcpip4_socket(host, port):
     try:
         s.connect((host, port))
         yield s
-    finally:
         s.shutdown(socket.SHUT_RDWR)
         s.close()
+    except:
```
Comment (dat) [inline ner/utils.py:18]: This does not achieve the same result that I'm aiming for. When some other piece of code invokes this and gets a resulting socket and throws an error (maybe not be related to the socket), then the error gets caught by the except clause and discarded away. The socket shutdown/close lines following "yield s" never get executed and the socket doesn't get cleaned up. This change a) defeats the purpose of this function and b) obscures any exception that should be propagated upwards.

Comment (nubela) [inline ner/utils.py:18]: Btw, I turned Stanford NER into a startup http://nerily.com, what do you
make of it? :) Cheers.

Steven Goh,
_M:_ +65 96634246
Founder, Spawt.

_Spawt is the easiest way to discover great things happening at places
nearby. *
*Download Spawt for your iPhone now at http://dl.spawt.me_

Comment (dat) [inline ner/utils.py:18]: Great. Hope you can commercialize it. As always code contributions back, here and to the Stanford NER itself, are always welcomed.

Cheers,
Dat

**Comment by dat** [comment]
I'm going to close this thread, since there's been no updated.

**Pull request closed without merging**